app/forms/questionnaire_form.py

```diff
@@ -584,19 +585,22 @@ def _get_error_id(id_: str) -> str:
 
 
 def _clear_detail_answer_field(
-    form_data: MultiDict, question_schema: QuestionSchemaType
+    form_data: ImmutableMultiDict | MultiDict, question_schema: QuestionSchemaType
 ) -> MultiDict[str, Any]:
     """
     Clears the detail answer field if the parent option is not selected
     """
+    mutable_form_data = (
+        MultiDict(form_data) if isinstance(form_data, ImmutableMultiDict) else form_data
+    )
+
     for answer in question_schema.get("answers", []):
         for option in answer.get("options", []):
             if "detail_answer" in option and option["value"] not in form_data.getlist(
                 answer["id"]
             ):
-                if isinstance(form_data, ImmutableMultiDict):
-                    form_data = MultiDict(form_data)
-                form_data[option["detail_answer"]["id"]] = ""
+                mutable_form_data[option["detail_answer"]["id"]] = ""
+
     return form_data
```

Comment:
you've switched the variable name to `mutable_form_data` but forgotten to return that instead